src/webgpu/api/validation/encoding/queries/resolveQuerySet.spec.ts

```diff
@@ -1,13 +1,94 @@
 export const description = `
-TODO:
-- invalid GPUQuerySet
-- firstQuery and/or queryCount out of range
-- invalid destination buffer
-- destinationOffset out of range
-- ?
+Validation tests for resolveQuerySet.
 `;
-
+import { poptions } from '../../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUConst } from '../../../../constants.js';
 import { ValidationTest } from '../../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
+
+export const kQueryCount = 2;
+
+g.test('resolveQuerySet,first_query_and_query_count')
+  .desc(
+    `
+Tests that resolve query set with invalid firstQuery and queryCount:
+- firstQuery and/or queryCount out of range
+  `
+  )
+  .params([
+    { firstQuery: 0, queryCount: kQueryCount },
+    { firstQuery: 0, queryCount: kQueryCount + 1 },
+    { firstQuery: 1, queryCount: kQueryCount },
+    { firstQuery: kQueryCount, queryCount: 1 },
+  ])
+  .fn(async t => {
+    const { firstQuery, queryCount } = t.params;
+
+    const querySet = t.device.createQuerySet({ type: 'occlusion', count: kQueryCount });
+    const destination = t.device.createBuffer({
+      size: kQueryCount * 8,
+      usage: GPUBufferUsage.QUERY_RESOLVE,
+    });
+
+    const encoder = t.device.createCommandEncoder();
+    encoder.resolveQuerySet(querySet, firstQuery, queryCount, destination, 0);
+
+    t.expectValidationError(() => {
+      encoder.finish();
+    }, firstQuery + queryCount > kQueryCount);
+  });
+
+g.test('resolveQuerySet,destination_buffer')
```

Comment:
Good catch, missed this in the initial plan. nit: `destination_usage` or `destination,usage` or something